src/pages/Animal/Feed/index.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useLayoutEffect } from 'react';
+import { setStatusBarBackgroundColor } from 'expo-status-bar';
+
+import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { FirebaseFirestoreTypes } from '@react-native-firebase/firestore';
+
+import HeaderLayout from '../../../layouts/HeaderLayout';
+
+import AnimalCard from '../../../components/AnimalCard';
+import InfiniteScroll from '../../../components/InfiniteScroll';
+
+import { Theme } from '../../../constants';
+
+import {
+  Age,
+  AgeNames,
+  Animal,
+  Sex,
+  SexNames,
+  Size,
+  SizeNames,
+} from '../../../types/animal';
+
+import { formatLocation } from '../../../utils/formatTexts';
+
+import {
+  CardText, CardTextContainer, CardTextRow, Container,
+} from './styles';
+
+// Service imports.
+import userAPI from '../../../services/user/api';
+import animalAPI from '../../../services/animal/api';
+
+const FeedPets = (): JSX.Element => {
+  const navigation = useNavigation();
+
+  useLayoutEffect(() => {
+    setStatusBarBackgroundColor(Theme.elements.statusBarSecondaryDark, false);
+  }, [navigation]);
+
+  const animalKey = (animalItem: Animal): string => animalItem.id;
+
+  const formatAnimal = (pet: Animal): JSX.Element => (
+    <AnimalCard
+      key={pet?.id}
+      imageUrl={pet?.pictures.length > 0 ? pet?.pictures[0] : null}
+      body={(
+        <CardTextContainer>
+          <CardTextRow>
+            <CardText>{SexNames[pet?.sex as Sex]}</CardText>
+            <CardText>{SizeNames[pet?.size as Size]}</CardText>
+            <CardText>{AgeNames[pet?.age as Age]}</CardText>
+          </CardTextRow>
+          <CardTextRow>
+            <CardText>
+              {formatLocation(pet?.ownerDocument)}
+            </CardText>
+          </CardTextRow>
+        </CardTextContainer>
+      )}
+      title={pet?.name}
+      headerOptions={(
+        <MaterialCommunityIcons
+          name="heart-outline"
+          size={24}
+          color={Theme.elements.headerText}
+        />
+      )}
+      headerBackground={Theme.elements.headerSecondary}
+      onPress={() => navigation.navigate('AnimalDetails', { animalUID: pet?.id })}
+    />
+  );
+
+  // Fetch pets function declaration
+  const fetchPets = (
+    lastElement: Animal | null, pageNumber: number, pageSize: number,
+  ): Promise<Array<Animal>> => {
+    let query : FirebaseFirestoreTypes.Query;
+    const orderBy = 'name';
+
+    if (pageNumber === 1 || lastElement == null) {
+      query = animalAPI.createQuery({
```

Comment:
Could we change this from createQuery to createQueryByUserUid to fetch only animals that don't belong to the user?

Reply:
Issue opened for future investigation.